sdk/src/main/java/com/sygic/travel/sdk/auth/facade/AuthFacade.kt

```diff
@@ -0,0 +1,128 @@
+package com.sygic.travel.sdk.auth.facade
+
+import android.util.Log
+import com.sygic.travel.sdk.auth.model.AuthorizationRequest
+import com.sygic.travel.sdk.auth.model.AuthorizationResponse
+import com.sygic.travel.sdk.auth.model.UserRegistrationResponse
+import com.sygic.travel.sdk.auth.service.AuthService
+import ru.gildor.coroutines.retrofit.Result
+import ru.gildor.coroutines.retrofit.awaitResult
+
+
+class AuthFacade(
+	private val authService: AuthService
+) {
+
+	suspend fun getUserSession(): AuthorizationResponse? {
+		val authRequest = AuthorizationRequest()
+		authRequest.clientId = CLIENT_ID
+		authRequest.grantType = GrantType.CLIENT_CREDENTIALS.grantType
+
+		val result = authService.authorizationQuery(authRequest).awaitResult()
```

Comment:
this imo should be moved to handleResult method.

Reply:
and rename that method to exectureRequest

Reply:
Well, this can be done only if we have one type of request, but in this case we have two:
* `authService.authorizationQuery(authRequest).awaitResult()`
* `authService.userRegistration( ... ).awaitResult()`
Therefore I would need to create two `exectureRequest()` methods.